lib/js/src/manager/screen/_ScreenManagerBase.js

```diff
@@ -489,6 +498,107 @@ class _ScreenManagerBase extends _SubManagerBase {
+        } else if (location === _ManagerLocation.ALERT_MANAGER) {
+            _ScreenManagerBase._softButtonIDByAlertManager.clear();
+        }
+
+        const buttonIdsSetHashSet = new Set();
+        let currentSoftButtonId,
+            numberOfButtonIdsSet = 0,
+            maxButtonIdsSetByDev = SOFT_BUTTON_ID_MIN_VALUE;
+
+        for (const softButtonObject of softButtonObjects) {
+            currentSoftButtonId = softButtonObject.getButtonId();
+            if (currentSoftButtonId !== SOFT_BUTTON_ID_NOT_SET_VALUE) {
+                if (_ScreenManagerBase._softButtonIDByAlertManager.has(currentSoftButtonId) || _ScreenManagerBase._softButtonIDBySoftButtonManager.has(currentSoftButtonId)) {
+                    return false;
+                }
+                numberOfButtonIdsSet++;
+                if (currentSoftButtonId > maxButtonIdsSetByDev) {
+                    maxButtonIdsSetByDev = currentSoftButtonId;
+                }
+                buttonIdsSetHashSet.add(softButtonObject.getButtonId());
+            }
+        }
+
+        if (numberOfButtonIdsSet !== buttonIdsSetHashSet.size) {
+            return false;
+        }
+
+        let generatedSoftButtonId = maxButtonIdsSetByDev;
+        for (const softButtonObject of softButtonObjects) {
+            currentSoftButtonId = softButtonObject.getButtonId();
+            if (currentSoftButtonId === SOFT_BUTTON_ID_NOT_SET_VALUE) {
+                while (buttonIdsSetHashSet.has(generatedSoftButtonId)) {
+                    if (generatedSoftButtonId >= SOFT_BUTTON_ID_MAX_VALUE) {
+                        generatedSoftButtonId = SOFT_BUTTON_ID_MIN_VALUE;
+                    }
+                    generatedSoftButtonId++;
+                }
+                softButtonObject._setButtonId(generatedSoftButtonId);
+                buttonIdsSetHashSet.add(generatedSoftButtonId);
+                if (location === _ManagerLocation.ALERT_MANAGER) {
+                    _ScreenManagerBase._softButtonIDByAlertManager.add(generatedSoftButtonId);
+                } else if (location === _ManagerLocation.SOFTBUTTON_MANAGER) {
+                    _ScreenManagerBase._softButtonIDBySoftButtonManager.add(generatedSoftButtonId);
+                }
+            }
+        }
+        return true;
+    }
+}
+
+class _ManagerLocation extends Enum {
+    constructor () {
+        super();
+    }
+
+    static get SOFTBUTTON_MANAGER () {
+        return _ManagerLocation._MAP.SOFTBUTTON_MANAGER;
+    }
+
+    static get ALERT_MANAGER () {
+        return _ManagerLocation._MAP.ALERT_MANAGER;
+    }
 }
 
+_ManagerLocation._MAP = Object.freeze({
+    'SOFTBUTTON_MANAGER': 0,
+    'ALERT_MANAGER': 1,
+});
+
+_ScreenManagerBase._ManagerLocation = _ManagerLocation;
+
+_ScreenManagerBase._softButtonIDBySoftButtonManager = new Set();
+_ScreenManagerBase._softButtonIDByAlertManager = new Set();
+
+const SOFT_BUTTON_ID_NOT_SET_VALUE = -1;
+const SOFT_BUTTON_ID_MIN_VALUE = 0;
```

Comment:
Not sure where to put this since I couldn't figure out the bug in time, but there's an issue with the OnButtonPress event not being sent for the first softbuttonobject set in an alertview, but the other ones work. If you set SOFT_BUTTON_ID_MIN_VALUE here to 1 then it will work. This leads me to believe that there is some button id check being done somewhere naively that would result in the first button id being 0 and treated as a falsy value and thus not handled correctly, but I'm not sure. Here's what I used:

```
const alertView = new SDL.manager.screen.utils.AlertView()
                .setText('hello')
                .setSoftButtons([
                new SDL.manager.screen.utils.SoftButtonObject('game', [state1], 'ROCK', (id, rpc) => {
                    if (rpc instanceof SDL.rpc.messages.OnButtonPress) {
                        console.log('First button pressed!');
                    }
                }),
                new SDL.manager.screen.utils.SoftButtonObject('button', [state4], 'BUTTON', (id, rpc) => {
                    if (rpc instanceof SDL.rpc.messages.OnButtonPress) {
                        console.log('Second button pressed!');
                    }
                }),
            ])

            console.log(await screenManager.presentAlert(alertView));const alertView = new SDL.manager.screen.utils.AlertView()
                .setText('hello')
                .setSoftButtons([
                new SDL.manager.screen.utils.SoftButtonObject('game', [state1], 'ROCK', (id, rpc) => {
                    if (rpc instanceof SDL.rpc.messages.OnButtonPress) {
                        console.log('First button pressed!');
                    }
                }),
                new SDL.manager.screen.utils.SoftButtonObject('button', [state4], 'BUTTON', (id, rpc) => {
                    if (rpc instanceof SDL.rpc.messages.OnButtonPress) {
                        console.log('Second button pressed!');
                    }
                }),
            ])

            console.log(await screenManager.presentAlert(alertView));
```

Reply:
I believe an issue like this was brought up before with the other teams, and the resolution was that the manager's SoftButton IDs should start at 1. To resolve this and to better align with the Java Suite's implementation, I've updated a while loop to a do while loop in the _ScreenManagerBase._checkAndAssignButtonIds method so the IDs always start at 1.